app/models/foreman_tasks/task/dynflow_task.rb

```diff
@@ -41,12 +41,18 @@ def execution_plan
       @execution_plan ||= ForemanTasks.dynflow.world.persistence.load_execution_plan(external_id)
     end
 
+    def label
```

Comment:
Could you put this logic into separate PR. Btw. I'm just looking into this code from performance perspective: I might try to reduce the the calls to laod teh execution plan from here.